File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TDBXWideStringValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXWideStringValue.h"


@implementation TDBXWideStringValue
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:WideStringType];
	}
	return self;
}
-(void) dealloc{
	[DBXStringValue release];
	[super dealloc];
}
-(void) SetNull {
	ValueNull = YES;
	[DBXStringValue release];
	DBXStringValue = @"";
	[DBXStringValue retain];
}

-(bool) isNull {
	return ValueNull;
}

-(void) SetAsString:(NSString*) value{
	ValueNull = NO;
	[DBXStringValue release];
	DBXStringValue = [value retain];
}

-(NSString*) GetAsString {
	return DBXStringValue;
}

@end

File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXWideStringValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * 
 * Wraps the WideString type and allows it to be null
 *
 */


public class TDBXWideStringValue extends DBXValue {
	protected boolean ValueNull = false;
	private String DBXWideStringValue;

	public TDBXWideStringValue() {
		super();
		setDBXType(DBXDataTypes.WideStringType);
	}
	
	public void setNull() {
		ValueNull = true;
		DBXWideStringValue = null;
	}

	public boolean isNull() {
		return ValueNull;
	}

	@Override
	public void SetAsString(String Value) throws DBXException {
		ValueNull = false;
		DBXWideStringValue = Value;
	}

	@Override
	public String GetAsString() throws DBXException {
		return DBXWideStringValue;
	}
}
